### PR TITLE
Reduce ErasedAst to FinalAst

### PR DESCRIPTION
**Pull request by JonathanStarup**
First two points of #2591

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala [hunk 1]
```diff
@@ -29,330 +27,180 @@ object ErasedAst {
...
-    final val tpe = EType.Char()
-  }
-
-  case class UnboxFloat32(exp: ErasedAst.Expression[PReference[PBoxedFloat32]], loc: SourceLocation) extends ErasedAst.Expression[PFloat32] {
-    final val tpe = EType.Float32()
-  }
-
-  case class UnboxFloat64(exp: ErasedAst.Expression[PReference[PBoxedFloat64]], loc: SourceLocation) extends ErasedAst.Expression[PFloat64] {
-    final val tpe = EType.Float64()
-  }
```
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala:355]: I would think they could be left in the AST.

[the remark this answers is not in the document]
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala:355]: I was intending to add it later. I can add them again to the Ast if you want. I didn't want to introduce casts in the eraser to delay any changes until erased Ast is actually used by the backend.

[the remark this answers is not in the document]
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala:355]: I intended to add them later. I am not sure what type to give them? Take tpe as an argument?

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala:355]: The relevant types? E.g. Int32 or ##java.lang.Integer?

**Pull request merged**